e2e/run_tests.js

```diff
@@ -3,13 +3,99 @@
 
 /* eslint-disable no-await-in-loop, no-console */
 
+const argv = require('yargs').argv;
 const axios = require('axios');
 const cypress = require('cypress');
 const fse = require('fs-extra');
 const {merge} = require('mochawesome-merge');
 const generator = require('mochawesome-report-generator');
 
-const MAX_FAILED_TITLES = 5;
+runTests();
+
+async function runTests() {
+    // Test directories that requires Cypress chromeWebSecurity disabled
+    const WITH_DISABLED_CHROME_WEB_SECURITY = ['enterprise_saml'];
+    const REPORT_DIR = 'results/mochawesome-report';
+
+    await fse.remove('results');
+    await fse.remove('screenshots');
+
+    const allDir = fse.readdirSync('cypress/integration/');
+
+    let testDir;
+    if (argv.dir) {
+        testDir = argv.dir.split(',').filter((d) => Boolean(d)).map((d) => d.trim());
+
+        const invalidDir = [];
+        testDir.forEach((dir) => {
+            if (!allDir.includes(dir)) {
+                invalidDir.push(dir);
+            }
+        });
+
+        if (invalidDir.length > 0) {
+            console.log(`Entered invalid directory: "${invalidDir}"`);
+            process.exit(1); // eslint-disable-line no-process-exit
+        }
+    } else {
+        testDir = allDir;
+    }
+
+    let failedTests = 0;
+
+    for (const dir of testDir) {
+        const disableChromeWebSecurity = WITH_DISABLED_CHROME_WEB_SECURITY.includes(dir);
+
+        const {totalFailed} = await cypress.run(getOptions({specDir: dir, disableChromeWebSecurity}));
+        failedTests += totalFailed;
+    }
+
+    // Merge all json reports into one single json report
+    const jsonReport = await merge({reportDir: REPORT_DIR});
+
+    // Generate short summary, write to file and then send report via webhook
+    const summary = generateShortSummary(jsonReport);
+    writeJsonToFile(summary, 'summary.json', REPORT_DIR);
+    await sendReport(summary);
+
+    // Generate the html report file
+    await generator.create(jsonReport, {reportDir: REPORT_DIR});
+
+    // eslint-disable-next-line
+    process.exit(failedTests); // exit with the number of failed tests
+}
+
+function getOptions({specDir, disableChromeWebSecurity}) {
+    const reporterDir = 'results/mochawesome-report';
+
+    const browser = disableChromeWebSecurity ? {browser: 'chrome'} : {};
```

Comment:
Our Cypress tests normally run on test server with electron/headless while the new SAML Okta requires chrome/headed.